client/src/controllers/SlugController.ts

```diff
@@ -14,9 +14,62 @@ export class SlugController extends Controller<HTMLInputElement> {
 
   declare allowUnicodeValue: boolean;
 
-  slugify() {
-    this.element.value = cleanForSlug(this.element.value.trim(), false, {
-      unicodeSlugsEnabled: this.allowUnicodeValue,
-    });
+  /**
+   * Allow for a comparison value to be provided, if does not compare to the
+   * current value (once transformed), then the event's default will
+   * be prevented.
+   */
+  compare(
+    event: CustomEvent<{ value: string }> & { params?: { compareAs?: string } },
+  ) {
+    // do not attempt to compare if the current field is empty
+    if (!this.element.value) {
+      return true;
+    }
+
+    const {
+      detail: { value = '' } = {},
+      params: { compareAs = 'slugify' } = {},
+    } = event;
+
+    const compareValue = this[compareAs]({ detail: { value } }, true);
+    const currentValue = this.element.value;
+
+    const valuesAreSame = compareValue.trim() === currentValue.trim();
+
+    if (!valuesAreSame) {
+      event?.preventDefault();
+    }
+
+    return valuesAreSame;
+  }
+
+  slugify(event: CustomEvent<{ value: string }>, ignoreUpdate = false) {
+    const unicodeSlugsEnabled = this.allowUnicodeValue;
+    const { value = this.element.value } = event?.detail || {};
+    const newValue = cleanForSlug(value.trim(), false, { unicodeSlugsEnabled });
+
+    if (!ignoreUpdate) {
+      this.element.value = newValue;
+    }
+
+    return newValue;
+  }
+
+  /**
+   * Advanced slugify of a string, updates the controlled element's value
+   * or can be used to simply return the transformed value.
+   * If a custom event with detail.value is provided, that value will be used
+   * instead of the field's value.
+   */
+  urlify(event: CustomEvent<{ value: string }>, ignoreUpdate = false) {
+    const { value = this.element.value } = event?.detail || {};
+    const newValue = cleanForSlug(value.trim(), true);
+
```

Comment:
Well, probably not for any practical scenario, however if you look at the code if the urlify does resolves to a non-empty string it will fall back on slugify which does need the param.

Reply:
True 👍